AnnoDesigner/App.xaml.cs

```diff
@@ -117,107 +132,98 @@ private async void Application_Startup(object sender, StartupEventArgs e)
-                    logger.Error(ex, "Error upgrading settings.");
+                    //mutex was killed
+                    createdNewMutex = true;
+                }
+            }
 
-                    _messageBoxService.ShowError("The settings file has become corrupted. We must reset your settings.");
+            try
+            {
+                //check if file is not corrupt
+                _appSettings.Reload();
 
-                    var fileName = "";
-                    if (!string.IsNullOrEmpty(ex.Filename))
-                    {
-                        fileName = ex.Filename;
-                    }
-                    else
-                    {
-                        var innerException = ex.InnerException as ConfigurationErrorsException;
-                        if (innerException != null && !string.IsNullOrEmpty(innerException.Filename))
-                        {
-                            fileName = innerException.Filename;
-                        }
-                    }
+                if (_appSettings.SettingsUpgradeNeeded)
+                {
+                    _appSettings.Upgrade();
+                    _appSettings.SettingsUpgradeNeeded = false;
+                    _appSettings.Save();
+                }
+            }
+            catch (ConfigurationErrorsException ex)
+            {
+                logger.Error(ex, "Error upgrading settings.");
+
+                _messageBoxService.ShowError("The settings file has become corrupted. We must reset your settings.");
 
-                    if (File.Exists(fileName))
+                var fileName = "";
+                if (!string.IsNullOrEmpty(ex.Filename))
+                {
+                    fileName = ex.Filename;
+                }
+                else
+                {
+                    if (ex.InnerException is ConfigurationErrorsException innerException && !string.IsNullOrEmpty(innerException.Filename))
                     {
-                        File.Delete(fileName);
+                        fileName = innerException.Filename;
                     }
-
-                    _appSettings.Reload();
                 }
 
-                //var updateWindow = new UpdateWindow();                
-                await _updateHelper.ReplaceUpdatedPresetsFilesAsync();
-
-                var recentFilesSerializer = new RecentFilesAppSettingsSerializer(_appSettings);
+                if (File.Exists(fileName))
+                {
+                    File.Delete(fileName);
+                }
 
-                IRecentFilesHelper recentFilesHelper = new RecentFilesHelper(recentFilesSerializer, _fileSystem);
-                var mainVM = new MainViewModel(_commons, _appSettings, recentFilesHelper, _messageBoxService, _updateHelper, _localizationHelper, _fileSystem);
+                _appSettings.Reload();
+            }
 
-                //TODO MainWindow.ctor calls AnnoCanvas.ctor loads presets -> change logic when to load data 
```

Comment:
I think this comment still reflects the current state.

Reply:
I'll get that added back in then